stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/coil/StreamImageLoaderFactory.kt

```diff
@@ -22,43 +22,36 @@ import coil.ImageLoader
 import coil.ImageLoaderFactory
 import coil.decode.GifDecoder
 import coil.decode.ImageDecoderDecoder
-import coil.util.CoilUtils
-import okhttp3.Dispatcher
-import okhttp3.Interceptor
-import okhttp3.OkHttpClient
+import coil.decode.VideoFrameDecoder
+import coil.disk.DiskCache
+import coil.memory.MemoryCache
 
 private const val DEFAULT_MEMORY_PERCENTAGE = 0.25
+private const val DEFAULT_DISK_CACHE_PERCENTAGE = 0.02
+private const val DISK_CACHE_DIRECTORY = "image_cache"
+
 public class StreamImageLoaderFactory(
     private val context: Context,
-    private val builder: ImageLoader.Builder.() -> Unit = {}
+    private val builder: ImageLoader.Builder.() -> Unit = {},
 ) : ImageLoaderFactory {
     override fun newImageLoader(): ImageLoader {
         return ImageLoader.Builder(context)
-            .availableMemoryPercentage(DEFAULT_MEMORY_PERCENTAGE)
+            .memoryCache { MemoryCache.Builder(context).maxSizePercent(DEFAULT_MEMORY_PERCENTAGE).build() }
             .allowHardware(false)
             .crossfade(true)
-            .okHttpClient {
-                val cacheControlInterceptor = Interceptor { chain ->
-                    chain.proceed(chain.request())
-                        .newBuilder()
-                        .header("Cache-Control", "max-age=3600,public")
-                        .build()
-                }
-                // Don't limit concurrent network requests by host.
-                val dispatcher = Dispatcher().apply { maxRequestsPerHost = maxRequests }
-
-                OkHttpClient.Builder()
-                    .cache(CoilUtils.createDefaultCache(context))
-                    .dispatcher(dispatcher)
-                    .addNetworkInterceptor(cacheControlInterceptor)
+            .diskCache {
+                DiskCache.Builder()
+                    .directory(context.cacheDir.resolve(DISK_CACHE_DIRECTORY))
+                    .maxSizePercent(DEFAULT_DISK_CACHE_PERCENTAGE)
```

Comment:
Coil 1.x relied on OkHttp's disk cache. Coil 2.x has its own disk cache and should not use OkHttp's Cache.

Reply:
Done: [6c3e7fa](https://github.com/GetStream/stream-chat-android/pull/3538/commits/6c3e7fa1e9dcb513d84867c78f3a3a6091b71a3a)